ERCS/erc-7699.md

```diff
@@ -0,0 +1,162 @@
+
+By extending the existing ERC-20 token standard with payment reference capabilities, this proposal will help bridge the gap between traditional finance systems and the world of decentralized finance, providing a more seamless experience for users, merchants, and service providers alike.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+Any contract complying with ERC-20 when extended with this ERC, MUST implement the following interface:
+```
+// The EIP-165 identifier of this interface is 0x1522573a
+
+interface IERC7699 {
+
+function transfer(address to, uint256 amount, bytes calldata paymentReference) external returns (bool);
+
+function transferFrom(address from, address to, uint256 amount, bytes calldata paymentReference) external returns (bool);
+
+event Transfer(address indexed from, address indexed to, uint256 amount, bytes indexed paymentReference);
+
+}
+```
+
+These `transfer` and `transferFrom` functions MUST emit `Transfer` event with paymentReference parameter.
+
+`paymentReference` parameter MAY be empty - example: `emit Transfer(From, To, amount, "");`
+
+`paymentReference` parameter is not limited in length by design, users are motivated to keep it short by calldata and log data gas costs.
+
+Transfers of 0 amount MUST be treated as normal transfers and fire the `Transfer` event.
+
+
+## Rationale
+
+### Parameter name
+
+The choice to name the added parameter paymentReference was made to align with traditional banking terminology, where payment references are widely used to associate and reconcile transactions with specific orders, invoices or other financial records.
+
+The paymentReference parameter name also helps to clearly communicate the purpose of the parameter and its role in facilitating the association and reconciliation of transactions. By adopting terminology that is well-established in the financial industry, the proposal aims to foster a greater understanding and adoption of the extended ERC-20 token standard.
+
+## Backwards Compatibility
+
+This extension is fully backwards compatible with the existing ERC-20 token standard. The new functions can be used alongside the existing transfer and transferFrom functions. Existing upgradable ERC-20 tokens can be upgraded to include the new functionality without impact on the storage layout; new ERC-20 tokens can choose to implement the payment reference features based on their specific needs.
+
+ERC-20 requires its `Transfer(address indexed _from, address indexed _to, uint256 _value) ` event to be emitted during transfers, thus there will be duplicitous data logged (from, to, amount) in two events.
+
+## Reference Implementation
+
+```
+// SPDX-License-Identifier: MIT
```

Comment:
I can't merge this until either: this license is changed to CC0-1.0, or the code is moved into the `../assets/erc-7699/` directory.

Reply:
Thx, updated